src/core/ModelVisitor.cpp

```diff
@@ -0,0 +1,28 @@
+/*
+ * ModelVisitor.cpp - visitors for automatable models
+ *
+ * Copyright (c) 2019-2019 Johannes Lorenz <j.git$$$lorenz-ho.me, $$$=@>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "ModelVisitor.h"
+
+ModelVisitor::~ModelVisitor() {}
+ConstModelVisitor::~ConstModelVisitor() {}
```

Comment:
If those constructors do nothing, you can use default constructors and possibly remove this file. However, please feel free to ignore this if you have any other intentions to keep this file.

Reply:
I'd normally do so, but if I write `~ModelVisitor() {}` into the header, clang's `-Wweak-vtables` warns me that the vtable will be placed into every translation unit unless it has at least one function in a cpp file. For `= default;`, the warning vanished, but this seems to be an error of the clang switch.